refactor(minifier)!: rename `CompressOptions::all_true`/`all_false` to `smallest`/`safest`

sapphi-red opened the pull request:
With the `keep_names` option, `all_true` will not behave as you would expect. I think what is expected is `smallest` / `safest` than `all_true` / `all_false`.

Without this change, `all_true` will drop `console` calls but keep the function names as-is.

sapphi-red commented:
* **#9867** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9867?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#9866** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9866?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/9866?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=9866).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Boshen commented:
### Merge activity

* **Mar 18, 6:32 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
refactor(minifier)!: rename `CompressOptions::all_true`/`all_false` to `smallest`/`safest` (#9866)

With the `keep_names` option, `all_true` will not behave as you would expect. I think what is expected is `smallest` / `safest` than `all_true` / `all_false`.

Without this change, `all_true` will drop `console` calls but keep the function names as-is.
